muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/fieldio/FileAnswerUtils.java

```diff
@@ -0,0 +1,151 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import fi.otavanopisto.muikku.dao.base.SystemSettingDAO;
+import fi.otavanopisto.muikku.model.base.SystemSetting;
+import fi.otavanopisto.muikku.plugins.workspace.dao.WorkspaceMaterialFileFieldAnswerFileDAO;
+import fi.otavanopisto.muikku.plugins.workspace.model.WorkspaceMaterialFileFieldAnswer;
+import fi.otavanopisto.muikku.plugins.workspace.model.WorkspaceMaterialFileFieldAnswerFile;
+
+public class FileAnswerUtils {
+  
+  public static final String SETTING_ENABLED = "fileanswers.storageEnabled";
+  public static final String SETTING_FILEFOLDER = "fileanswers.storageFolder";
+  public static final String SETTING_LASTID = "fileanswers.lastId";
+
+  @Inject
+  private Logger logger;
+
+  @Inject
+  private SystemSettingDAO systemSettingDAO;
+  
+  @Inject
+  private WorkspaceMaterialFileFieldAnswerFileDAO workspaceMaterialFileFieldAnswerFileDAO;
+  
+  public Long getLastEntityId() {
+    SystemSetting idSetting = systemSettingDAO.findByKey(SETTING_LASTID);
+    return idSetting == null ? 0L : Long.valueOf(idSetting.getValue());
+  }
+  
+  public void setLastEntityId(Long id) {
+    SystemSetting idSetting = systemSettingDAO.findByKey(SETTING_LASTID);
+    if (idSetting == null) {
+      systemSettingDAO.create(SETTING_LASTID, id.toString());
+    }
+    else {
+      systemSettingDAO.update(idSetting, id.toString());
+    }
+  }
+  
+  public boolean isFileSystemStorageEnabled() {
+    SystemSetting enabledSetting = systemSettingDAO.findByKey(SETTING_ENABLED);
+    return enabledSetting != null && Boolean.valueOf(enabledSetting.getValue());
+  }
+  
+  public byte[] getFileContent(Long userEntityId, String fileId) throws IOException {
+    if (userEntityId == null || fileId == null) {
+      logger.severe(String.format("Invalid fetch parameters: file %s of user %d", fileId, userEntityId));
+      return new byte[0];
+    }
+    SystemSetting fileFolderSetting = systemSettingDAO.findByKey(SETTING_FILEFOLDER);
+    if (fileFolderSetting != null && !StringUtils.isBlank(fileFolderSetting.getValue())) {
+      java.io.File answerFile = Paths.get(fileFolderSetting.getValue(), userEntityId.toString(), fileId).toFile();
+      if (answerFile.exists()) {
+        return Files.readAllBytes(answerFile.toPath());
+      }
+    }
+    logger.warning(String.format("Failed to retrieve file %s of user %d", fileId, userEntityId));
+    return new byte[0];
+  }
+  
+  public void storeFileToFileSystem(Long userEntityId, String fileId, byte[] content) throws IOException {
+    if (userEntityId == null || fileId == null || content == null) {
+      logger.severe(String.format("Invalid store parameters or no file content: file %s of user %d", fileId, userEntityId));
+      return;
```

Comment:
Probably needs to be exception here to indicate the file couldn't be saved at all.

Reply:
Changed calling side to catch all exceptions